apps/omg_watcher_info/lib/omg_watcher_info/api/stats.ex

```diff
@@ -20,49 +20,58 @@ defmodule OMG.WatcherInfo.API.Stats do
   alias OMG.WatcherInfo.DB.Block
   alias OMG.WatcherInfo.DB.Transaction
 
+  @seconds_in_twenty_four_hours 86_400
+
   @doc """
   Retrieves network statistics.
   """
   def get() do
-    timestamps_all_time = Block.get_timestamps()
-    timestamps_last_24_hours = Block.get_timestamps_last_24_hours()
+    end_datetime = DateTime.to_unix(DateTime.utc_now())
+    start_datetime_24_hours = end_datetime - @seconds_in_twenty_four_hours
+
+    timestamps_all_time = Block.all_timestamps()
+    timestamps_24_hours = Block.all_timestamps_between(start_datetime_24_hours, end_datetime)
 
     response = %{
       transaction_count: %{
-        all_time: Transaction.get_count(),
-        last_24_hours: Transaction.get_count_last_24_hour()
+        all_time: Transaction.count_all(),
+        last_24_hours: Transaction.count_all_timestamp_between(start_datetime_24_hours, end_datetime)
       },
       block_count: %{
-        all_time: Block.get_count(),
-        last_24_hours: Block.get_count_last_24_hour()
+        all_time: Block.count_all(),
+        last_24_hours: Block.count_all_timestamp_between(start_datetime_24_hours, end_datetime)
       },
-      average_block_interval: %{
+      average_block_interval_seconds: %{
         all_time: get_average_block_interval(timestamps_all_time),
-        last_24_hours: get_average_block_interval(timestamps_last_24_hours)
+        last_24_hours: get_average_block_interval(timestamps_24_hours)
       }
     }
 
     {:ok, response}
   end
 
-  @spec get_average_block_interval([%{timestamp: integer}]) :: float | String.t()
+  @doc """
+  Calculates the average of the differences between timestamps.
+  """
+  @spec get_average_block_interval([%{timestamp: non_neg_integer()}]) :: float | String.t()
   def get_average_block_interval(timestamps) do
-    case length(timestamps) do
-      n when n < 2 ->
-        String.to_atom("N/A")
+    case timestamps do
+      [_, _ | _] ->
```

Comment:
this case is a bit sub-optimal. 
for example:
```
[first, a | b]  = [1,2,3,4]
[1, 2, 3, 4]
iex(4)> a
2
iex(5)> b
[3, 4]
```

Reply:
but I didn't measure this so it might be just a slight difference (for the better or worse).

```
def get_average_block_interval([%{timestamp: first_timestamp} | timestamps] = all_timestamps) do
  %{timestamp: last_timestamp} = List.last(timestamps)
  (last_timestamp - first_timestamp) / (length(all_timestamps) - 1)
end
def get_average_block_interval(_), do: nil
```

Reply:
It's used to detect that the list has at least 2 items though. Otherwise we can't find a time difference.

Reply:
Noted thanks @Ino! As @unnawut notes this will not pose a problem for this purpose specifically.

Reply:
I don't understand the connection... 
`[_|_]` is pattern matching a non empty list. Head | rest. 
in your pattern you're matching first, second and the rest... why?

Reply:
ahaaa, you need two. lol. brain fart!

Reply:
Also misunderstood. The above is what we are looking for! Thanks both.